src/lib/Libifl/pbsD_connect.c

```diff
@@ -428,7 +429,8 @@ dealloc_conn_list_single(int parentfd)
 
 	conn_list = pthread_getspecific(psi_key);
 	for (iter_conns = conn_list; iter_conns != NULL; prev = iter_conns, iter_conns = iter_conns->next) {
-		if (iter_conns->conn_arr[0].sd == parentfd) {
+		if ((iter_conns->conn_arr[0].sd == parentfd) ||
+		    (parentfd == -1 && !iter_conns->next)) {
```

Comment:
im not sure i understand this change. it seems like we want to delete iter_conns[0] when parentfd is -1, but only when there's a single set of connections. Can you please explain why this is needed?